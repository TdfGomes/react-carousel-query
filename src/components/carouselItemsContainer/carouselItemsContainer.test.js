/* eslint-disable react/prop-types */
import React from 'react';
import { render } from '@testing-library/react';
import CarouselItemsContainer from './carouselItemsContainer';

jest.mock('@hooks/useQueryManager', () => ({
  useQueryManager: () => ({
    currentIndex: 0,
    total: 2,
    items: [
      { id: 1, name: 'first' },
      { id: 2, name: 'second' },
    ],
    next: jest.fn(),
    previous: jest.fn(),
  }),
}));

describe('<CarouselItemsContainer />', () => {
  const ItemFn = ({ item }) => <p>{item.name}</p>;
  const BadgeFn = ({ currentIndex, total }) => {
    return <span>{`${currentIndex}/${total}`}</span>;
  };

  const props = {
    renderItem: ItemFn,
    renderBadge: BadgeFn,
  };

  test('It should render', () => {
    const { container, getByText } = render(
      <CarouselItemsContainer {...props} />,
    );
    expect(getByText('1/2')).toBeTruthy();
    expect(getByText('first')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('It should hide index', () => {
    const { container, getByText, queryByText } = render(
      <CarouselItemsContainer hideIndex {...props} />,
    );
    expect(queryByText('1/2')).toBeFalsy();
    expect(getByText('first')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
