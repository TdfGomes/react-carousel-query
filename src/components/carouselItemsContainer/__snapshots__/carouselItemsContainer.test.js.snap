// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselItemsContainer /> It should hide index 1`] = `
<div>
  <button
    class="arrowButton left"
    data-testid="button-left"
  >
    <svg
      fill="none"
      fill-rule="evenodd"
      height="15px"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      viewBox="0 0 8 5"
      width="40px"
    >
      <polyline
        points="1 1 3.77984472 4 6.53846154 1"
      />
    </svg>
  </button>
  <div
    class="container flexDisplay alignCenter fullHeight"
    style="transform: translateX(0px); transition-duration: 0s;"
  >
    <div
      class="container fullHeight fullWidth fullMaxHeight fullMinWidth"
    >
      <p>
        first
      </p>
    </div>
    <div
      class="container fullHeight fullWidth fullMaxHeight fullMinWidth"
    >
      <p>
        second
      </p>
    </div>
  </div>
  <button
    class="arrowButton right"
    data-testid="button-right"
  >
    <svg
      fill="none"
      fill-rule="evenodd"
      height="15px"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      viewBox="0 0 8 5"
      width="40px"
    >
      <polyline
        points="1 1 3.77984472 4 6.53846154 1"
      />
    </svg>
  </button>
</div>
`;

exports[`<CarouselItemsContainer /> It should render 1`] = `
<div>
  <span>
    1/2
  </span>
  <button
    class="arrowButton left"
    data-testid="button-left"
  >
    <svg
      fill="none"
      fill-rule="evenodd"
      height="15px"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      viewBox="0 0 8 5"
      width="40px"
    >
      <polyline
        points="1 1 3.77984472 4 6.53846154 1"
      />
    </svg>
  </button>
  <div
    class="container flexDisplay alignCenter fullHeight"
    style="transform: translateX(0px); transition-duration: 0s;"
  >
    <div
      class="container fullHeight fullWidth fullMaxHeight fullMinWidth"
    >
      <p>
        first
      </p>
    </div>
    <div
      class="container fullHeight fullWidth fullMaxHeight fullMinWidth"
    >
      <p>
        second
      </p>
    </div>
  </div>
  <button
    class="arrowButton right"
    data-testid="button-right"
  >
    <svg
      fill="none"
      fill-rule="evenodd"
      height="15px"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      viewBox="0 0 8 5"
      width="40px"
    >
      <polyline
        points="1 1 3.77984472 4 6.53846154 1"
      />
    </svg>
  </button>
</div>
`;
